main.go

```diff
@@ -0,0 +1,367 @@
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+// parseQueryParams returns the start and end values, or an error.
+func parseQueryParams(values url.Values) (int64, int64, error) {
+	start := values.Get("start")
+	end := values.Get("end")
+	if start == "" {
+		return 0, 0, errors.New("missing start parameter")
+	}
+	if end == "" {
+		return 0, 0, errors.New("missing end parameter")
+	}
+	startInt, err := strconv.ParseInt(start, 10, 64)
+	if err != nil || startInt < 0 {
+		return 0, 0, errors.New("invalid start parameter")
+	}
+	endInt, err := strconv.ParseInt(end, 10, 64)
+	if err != nil || endInt < 0 {
+		return 0, 0, errors.New("invalid end parameter")
+	}
+	if endInt <= startInt {
+		return 0, 0, errors.New("end must be greater than start")
+	}
+	return startInt, endInt, nil
+}
+
+// tile represents important numbers about a tile: where it starts, where it ends, its size,
+// and what CT backend URL it exists on (or is anticipated to exist on), and what s3 prefix
+// it should be stored/retrieved under.
+type tile struct {
+	start    int64
+	end      int64
```

Comment:
Probably worth being explicit in the struct comment that `end` is an _exclusive_ bound.

Reply:
oop, I documented that this is a half-open interval "like CT" but then I realized CT uses a closed interval (will return the entry for `end`). Gonna check some code real quick..

Reply:
Alright, updated the comments and implementation. The current vibe is that half-open intervals are better (easier to reason about, fewer +1/-1 adjustments), so we use them internally, and only convert to and from closed intervals when parsing the query and when generating the outbound URL for requests. Open to feedback that it would be simpler the other way around (i.e. just use closed intervals internally because that's what CT does).